lib/glific/third_party/bhasini/bhasini.ex

```diff
@@ -74,19 +188,28 @@ defmodule Glific.Bhasini do
   # locally before uploading it to GCS to get public URL of file to be used at flow level
   @spec download_encoded_file(map(), String.t()) :: String.t()
   defp download_encoded_file(response, uuid) do
+    pipeline_response =
+      get_in(response, ["pipelineResponse"])
+      |> Enum.filter(fn response -> response["taskType"] == "tts" end)
+
     encoded_audio =
-      get_in(response, [
-        "pipelineResponse",
-        Access.at(0),
-        "audio",
-        Access.at(0),
-        "audioContent"
-      ])
+      get_in(pipeline_response, [Access.at(0), "audio", Access.at(0), "audioContent"])
 
     decoded_audio = Base.decode64!(encoded_audio)
-
     path = System.tmp_dir!() <> "#{uuid}.mp3"
     :ok = File.write!(path, decoded_audio)
     path
   end
+
+  @doc """
+  This function validates supported languages in Glific before sending to Bhasini
+  """
+  @spec valid_language?(String.t(), String.t()) :: boolean()
+  def valid_language?(source_language, target_language) do
+    valid_languages = Map.keys(@language_codes)
+
+    if source_language in valid_languages and target_language in valid_languages,
```

Comment:
Returning this statement `source_language in valid_languages and target_language in valid_languages` is enough right here?